### PR TITLE
fix: Fix linter issues

### PR DESCRIPTION
**Pull request by martincik**
# Relates to:
*No specific issue linked - appears to be a code quality improvement PR*

# Risks
Low - Changes are purely related to code style and import organization with no functional changes.

# Background
## What does this PR do?
This PR addresses linting issues in the codebase by:
1. Reorganizing and consolidating imports
2. Removing unused imports
3. Removing empty catch variable declarations
4. Fixing import ordering
5. Removing extra blank lines

## What kind of change is this?
Improvements (misc. changes to existing features)

# Documentation changes needed?
My changes do not require a change to the project documentation.

# Testing
## Where should a reviewer start?
Review the changes in the following files:
- packages/core/src/embedding.ts
- packages/core/src/generation.ts  
- packages/core/src/parsing.ts

## Detailed testing steps
No, automated tests are fine. These changes are purely syntactic and don't affect functionality.

The changes include:
1. Import reorganization
2. Removal of unused imports (fs)
3. Simplifying catch blocks by removing unused error variables
4. Consistent import ordering

Since these are linter-driven changes, existing tests should cover functionality verification.

### CODE REVIEW

File: packages/core/src/parsing.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export function parseJsonArrayFromText(text: string) {
     if (jsonBlockMatch) {
         try {
             jsonData = JSON.parse(jsonBlockMatch[1]);
-        } catch (e) {
+        } catch {
```
Comment (monilpat) [inline packages/core/src/parsing.ts:68]: Can we log the error with console.error instead of removing this?

File: packages/core/src/parsing.ts [hunk 2]
```diff
@@ -118,7 +118,7 @@ export function parseJSONObjectFromText(
         if (objectMatch) {
             try {
                 jsonData = JSON.parse(objectMatch[0]);
-            } catch (e) {
+            } catch {
```
Comment (monilpat) [inline packages/core/src/parsing.ts:121]: same here and everywhere else

**Comment by monilpat** [comment]
LGTM outside of adding console.error for the catch statements

**Comment by martincik** [comment]
@jkbrooks Build and lint fixed.

**Pull request merged**